src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs

```diff
@@ -393,6 +402,39 @@ public void CopyTo(T[] array, int arrayIndex)
             Array.Copy(_items, 0, array, arrayIndex, _size);
         }
 
+        public void CopyTo(int sourceIndex, int count, Span<T> destination)
+        {
+            if (sourceIndex < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(sourceIndex), SR.ArgumentOutOfRange_NeedNonNegNum);
+            }
+
+            if (count < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.count, ExceptionResource.ArgumentOutOfRange_NeedNonNegNum);
+            }
+
+            if (_size - sourceIndex < count)
+            {
+                ThrowHelper.ThrowArgumentException(ExceptionResource.Argument_InvalidOffLen);
+            }
+
+            if (count > destination.Length)
+            {
+                throw new ArgumentException(SR.Argument_DestinationTooShort, nameof(destination));
+            }
+
+            if (_size > 0 && count > 0)
```

Comment:
I guess I don't to check size > 0. If count is 1, size should be 1+, or else it would throw in if (_size - sourceIndex < count)

Reply:
The standard we code most libraries to is that race condition in the app should not lead to buffer overrun. It means that you cannot depend on `_size <= _items.Length`. They can get out of sync if there is a race condition.

It would be best to just call `_items.AsSpan(sourceIndex, count).CopyTo(destination)` here. It will be easy to see that there is no buffer overrun that way.